lib/spack/spack/test/util/unparse/unparse.py

```diff
@@ -512,3 +577,12 @@ def test_async_with():
 @pytest.mark.skipif(sys.version_info < (3, 5), reason="Not supported < 3.5")
 def test_async_with_as():
     check_ast_roundtrip(async_with_as)
+
+
+@pytest.mark.skipif(sys.version_info < (3, 10), reason="Not supported < 3.10")
```

Comment:
3.10?

Reply:
Yes, https://docs.python.org/3.10/whatsnew/3.10.html#pep-634-structural-pattern-matching We started failing in 3.11 because we have a test (without assertion, see #33722 ) that was testing the core libs - and core libraries started using pattern matching in 3.11

Reply:
yeah, so skipif < (3, 11)?

Reply:
This would be the quick fix but IMO it’s good to make sure unparsing con you need to be supported across versions for package hash. It’ll bite us one day if not.

Reply:
@haampie This was an issue for unparsing also for 3.10 We just didn't realize it. It will be an issue for Spack when we support 3.10+